### ISSUE TITLE
Give advice on which STL container to use for which application?

### ISSUE THREAD

**Issue by hageboeck** [hsf-training/cpluspluscourse#448]
We mention containers, but we don't really say when you should use those. Is it wise to discuss this in the advanced course?
Should we even create an exercise, or write something on quickbench that we can show in the slides?

**Comment by bernhardmgruber** [comment]
I would say that most of the interesting containers are actually not in the STL (yet) :) The goto is `std::vector`. I don't think there are many good cases for `std::list` etc. We could show a benchmark here, but we could also just tell the students they should not use it :) `std::deque` is useful if you want iterator and reference stability when resizing, but then we need to discuss this. We could throw in `std::array` maybe.

More interesting containers in my option would be e.g. `boost::static_vector<T, N>` , `boost::small_vector<T>` or `boost::circular_buffer`. There is also `std::hive` coming up (a bucket array). 

So bottom line: I would not know what to put into such an exercise. We could make a container exercise though, that just uses a bunch of vectors.
